src/ansys/geometry/core/designer/body.py

```diff
@@ -788,6 +816,31 @@ def rotate(
             )
         )
 
+    @protect_grpc
+    @check_input_types
+    @reset_tessellation_cache
+    def get_collision(self, body: "Body") -> CollisionType:
```

Comment:
```suggestion
    def get_collision(self, body: "Body") -> CollisionType:
```

You cannot call the ``check_input_types`` header here since the class "Body" is not defined at this point. Also, you do not need to reset the tessellation with this operation since you are not modifying the underlying model.